src/main/java/org/jboss/tools/intellij/openshift/actions/component/DebugComponentAction.java

```diff
@@ -99,12 +100,17 @@ private void executeDebug(Project project, Component component, Odo odo, String
                 // run odo debug if not already running
                 if (checkOdoDebugNotRunning(odo, projectName, applicationName, component)) {
                     ProgressManager.getInstance().run(
-                            new Task.Modal(project, "Starting odo", false) {
+                            new Task.Modal(project, "Starting odo", true) {
                                 public void run(@NotNull ProgressIndicator indicator) {
                                     indicator.setText("Starting debugger session for the component "
-                                            + component.getName());
+                                            + component.getName() + ".");
                                     indicator.setIndeterminate(true);
                                     try {
+                                        if (ComponentKind.DEVFILE.equals(component.getInfo().getComponentKind())) {
+                                            indicator.setText2("Please wait while pushing the component with debug flag...");
```

Comment:
I think the text is too much technical, something like:

Please wait while component is switched to debug mode

seems more user friendly

Reply:
modified.